node.go

```diff
@@ -38,29 +38,18 @@ type Node interface {
 	// Returns whether the parquet column is required.
 	Required() bool
 
-	// Returns the number of child nodes.
-	//
-	// The method returns zero on leaf nodes.
-	NumChildren() int
+	// Returns true if this a leaf node.
+	Leaf() bool
 
-	// Returns the sorted list of child node names.
-	//
-	// The method returns an empty slice on leaf nodes.
+	// Returns a mapping of the node's fields.
 	//
-	// As an optimization, the returned slice may be the same across calls to
-	// this method. Applications should treat the return value as immutable.
-	ChildNames() []string
-
-	// Returns the child node associated with the given name, or nil if the
-	// name did not exist.
+	// As an optimization, the same slices may be returned by multiple calls to
+	// this method, programs must treat the returned values as immutable.
 	//
-	// The method panics if it is called on a leaf node.
-	ChildByName(name string) Node
-
-	// ValueByName is returns the sub-value with the given name in base.
-	ValueByName(base reflect.Value, name string) reflect.Value
+	// This method returns an empty mapping when called on leaf nodes.
+	Fields() []Field
 
-	// Returns the list of encodings used by the node and its children.
```

Comment:
I changed this requirement since it was the source of significant code complexity but was not useful: only leaf columns are encoded in parquet, there is no point exposing encodings for non-leaf columns.